corehq/apps/users/views/__init__.py

```diff
@@ -957,6 +986,28 @@ def undo_remove_web_user(request, domain, record_id):
         reverse(ListWebUsersView.urlname, args=[domain]))
 
 
+@always_allow_project_access
+@require_can_edit_web_users
+@require_POST
+@location_safe
+def deactivate_web_user(request, domain, couch_user_id):
+    user = WebUser.get_by_user_id(couch_user_id, domain)
+    user.deactivate(domain, changed_by=request.couch_user)
```

Comment:
I may have missed it - is there any validation that prevents the user from deactivating themselves? If not, I think that can be added here and then on the frontend, don't display the "deactivate" button for the request user.

Reply:
The deactivate/reactivate buttons already don't show/aren't initialized for the current user.